dyninstAPI/src/AmdgpuKernelDescriptor.C

```diff
@@ -0,0 +1,1011 @@
+  return (amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX700 &&
+          amdgpuMach <= EF_AMDGPU_MACH_AMDGCN_GFX704) ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX705;
+}
+
+bool AmdgpuKernelDescriptor::isGfx8() const {
+  return (amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX801 &&
+          amdgpuMach <= EF_AMDGPU_MACH_AMDGCN_GFX810) ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX805;
+}
+
+bool AmdgpuKernelDescriptor::isGfx9() const {
+  return isGfx90aOr942() ||
+         (amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX900 && amdgpuMach <= EF_AMDGPU_MACH_AMDGCN_GFX90C);
+}
+
+bool AmdgpuKernelDescriptor::isGfx90aOr942() const {
+  return amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX90A && amdgpuMach <= EF_AMDGPU_MACH_AMDGCN_GFX942;
+}
+
+bool AmdgpuKernelDescriptor::isGfx9Plus() const { return isGfx9() || isGfx10Plus(); }
+
+bool AmdgpuKernelDescriptor::isGfx10() const {
+  return (amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX1010 &&
+          amdgpuMach <= EF_AMDGPU_MACH_AMDGCN_GFX1033) ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1013 ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1034 ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1035 || amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1036;
+}
+
+bool AmdgpuKernelDescriptor::isGfx10Plus() const { return isGfx10() || isGfx11(); }
+
+bool AmdgpuKernelDescriptor::isGfx11() const {
+  return amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1100 ||
+         amdgpuMach == EF_AMDGPU_MACH_AMDGCN_GFX1103 ||
+         (amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX1101 &&
+          amdgpuMach >= EF_AMDGPU_MACH_AMDGCN_GFX1102);
+}
+
+AmdgpuKernelDescriptor::AmdgpuKernelDescriptor(uint8_t *kdBytes, size_t kdSize, unsigned amdgpuMachine) {
+  assert(kdSize == sizeof(kernel_descriptor_t));
+  kdRepr = read_memory_as<kernel_descriptor_t>(kdBytes);
+  amdgpuMach = amdgpuMachine;
+}
+
+uint32_t AmdgpuKernelDescriptor::getGroupSegmentFixedSize() const {
+  return kdRepr.group_segment_fixed_size;
+}
+
+void AmdgpuKernelDescriptor::setGroupSegmentFixedSize(uint32_t value) {
+  kdRepr.group_segment_fixed_size = value;
+}
+
+uint32_t AmdgpuKernelDescriptor::getPrivateSegmentFixedSize() const {
+  return kdRepr.private_segment_fixed_size;
+}
+
+void AmdgpuKernelDescriptor::setPrivateSegmentFixedSize(uint32_t value) {
+  kdRepr.private_segment_fixed_size = value;
+}
+
+uint32_t AmdgpuKernelDescriptor::getKernargSize() const { return kdRepr.kernarg_size; }
+
+void AmdgpuKernelDescriptor::setKernargSize(uint32_t value) { kdRepr.kernarg_size = value; }
+
+int64_t AmdgpuKernelDescriptor::getKernelCodeEntryByteOffset() const {
+  return kdRepr.kernel_code_entry_byte_offset;
+}
+
+void AmdgpuKernelDescriptor::setKernelCodeEntryByteOffset(int64_t value) {
+  kdRepr.kernel_code_entry_byte_offset = value;
+}
+
+#define GET_VALUE(MASK) ((fourByteBuffer & MASK) >> (MASK##_SHIFT))
+#define SET_VALUE(MASK) (fourByteBuffer | ((value) << (MASK##_SHIFT)))
+#define CLEAR_BITS(MASK) (fourByteBuffer & (~(MASK)))
```

Comment:
`CLEAR_BITS` should always be called before `SET_VALUE` (and is), so `CLEAR_BITS` should be removed and be incorporated into `SET_VALUE`